File: src/3033.modify-the-matrix.3330/solution.ts
function modifiedMatrix(matrix: number[][]): number[][] {
    const M = matrix.length;
    const N = matrix[0].length;
    const cols = new Array<number>(N).fill(-1);
    const result = new Array<number[]>(M);
    const points: number[][] = [];
    for (let i = 0; i < M; i++) {
        result[i] = new Array<number>(N);
        for (let j = 0; j < N; j++) {
            if (matrix[i][j] === -1) {
                points.push([i, j])
            } else {
                result[i][j] = matrix[i][j];
                cols[j] = Math.max(cols[j], matrix[i][j])
            }
        }
    }
    for (const [x, y] of points) {
        result[x][y] = cols[y];
    }
    return result
};

File: src/3033.modify-the-matrix.3330/solution.java
class Solution {
    public int[][] modifiedMatrix(int[][] matrix) {
        int M = matrix.length;
        int N = matrix[0].length;
        int[] maxs = new int[N];
        for (int[] ints : matrix) {
            for (int j = 0; j < N; j++) {
                maxs[j] = Math.max(maxs[j], ints[j]);
            }
        }
        for(int i=0;i<M;i++){
            for(int j=0;j<N;j++){
                if(matrix[i][j] == -1){
                    matrix[i][j] = maxs[j];
                }
            }
        }
        return matrix;
    }
}